crates/driver/src/infra/blockchain/token.rs

```diff
@@ -111,29 +116,42 @@ impl Erc20 {
         source: order::SellTokenBalance,
         interactions: &[eth::Interaction],
     ) -> Result<eth::TokenAmount, Error> {
+        let mut method = self.balances.balance(
+            (
+                self.balances.address(),
+                self.vault_relayer.into(),
+                self.vault.into(),
+            ),
+            trader.into(),
+            self.token.address(),
+            0.into(),
+            ethcontract::Bytes(source.hash().0),
+            interactions
+                .iter()
+                .map(|i| {
+                    (
+                        i.target.into(),
+                        i.value.into(),
+                        ethcontract::Bytes(i.call_data.0.clone()),
+                    )
+                })
+                .collect(),
+        );
+        // Create the access list for the balance simulation
+        if let Some(from) = &method.tx.from {
+            let access_list_tx = Tx {
+                from: from.address(),
+                to: method.tx.to,
+                value: method.tx.value,
+                data: method.tx.data.clone(),
+                access_list: method.tx.access_list,
+            };
+            let access_list = self.ethereum.create_access_list(access_list_tx).await.ok();
```

Comment:
The `Balances` contract is not actually deployed so this `create_access_list` call would do nothing useful. Instead you'd have to compute the access list for the call that `contracts::storage_accessible::simulate()` executes under the hood.